src/content/symbol-store-db.js

```diff
@@ -1,134 +1,203 @@
+        transaction.onerror = () => reject(transaction.error);
+        const store = transaction.objectStore('symbol-tables');
+        this._deleteLeastRecentlyUsedUntilCountIsNoMoreThanN(store, this._maxCount - 1, () => {
+          const lastUsedDate = new Date();
+          const addReq = store.add({ debugName, breakpadId, addrs, index, buffer, lastUsedDate });
+          addReq.onsuccess = () => resolve();
+        });
+      });
     });
   }
 
-  getLibKey(debugName, breakpadId) {
-    if (!this._db) {
-      return this._setupDBPromise.then(() => this.getLibKey(debugName, breakpadId));
-    }
-
-    return new Promise((resolve, reject) => {
-      const transaction = this._db.transaction('symbol-tables', 'readonly');
-      const store = transaction.objectStore('symbol-tables');
-      const index = store.index('libKey');
-      const req = index.openKeyCursor(IDBKeyRange.only([debugName, breakpadId]));
-      req.onerror = reject;
-      req.onsuccess = () => {
-        const cursor = req.result;
-        if (cursor) {
-          resolve(cursor.primaryKey);
-        } else {
-          reject(new Error('Nothing in the async cache for this library.'));
-        }
-      };
+  /**
+   * Retrieve the symbol table for the given library.
+   * @param {string}      The debugName of the library.
+   * @param {string}      The breakpadId of the library.
+   * @return              A promise that resolves with the symbol table (in
+   *                      SymbolTableAsTuple format), or fails if we couldn't
+   *                      find a symbol table for the requested library.
+   */
+  getSymbolTable(debugName: string, breakpadId: string): Promise<SymbolTableAsTuple> {
+    return this._dbPromise.then(db => {
+      return new Promise((resolve, reject) => {
+        const transaction = db.transaction('symbol-tables', 'readwrite');
+        transaction.onerror = () => reject(transaction.error);
+        const store = transaction.objectStore('symbol-tables');
+        const index = store.index('libKey');
+        const req = index.openCursor(window.IDBKeyRange.only([debugName, breakpadId]));
+        req.onsuccess = () => {
+          const cursor = (req.result: any);
+          if (cursor) {
+            const value = (cursor: IDBCursorWithValue).value;
+            value.lastUsedDate = new Date();
+            const updateDateReq = store.put(value, cursor.primaryKey);
+            const { addrs, index, buffer } = value;
+            updateDateReq.onsuccess = () => resolve([addrs, index, buffer]);
+          } else {
+            reject(new Error('The requested library does not exist in the database.'));
+          }
+        };
+      });
     });
   }
 
-  importLibrary(debugName, breakpadId, [addrs, index, buffer]) {
-    if (!this._db) {
-      return this._setupDBPromise.then(() => this.importLibrary(debugName, breakpadId, [addrs, index, buffer]));
-    }
-
+  _setupDB(dbName: string): Promise<IDBDatabase> {
```

Comment:
nit: I'd put the setupDB method first after the constructor, so that when I read the file linearly I get the DB structure first.